python/pyspark/pandas/indexes/timedelta.py

```diff
@@ -111,3 +126,80 @@ def __getattr__(self, item: str) -> Any:
                 return partial(property_or_func, self)
 
         raise AttributeError("'TimedeltaIndex' object has no attribute '{}'".format(item))
+
+    @property
+    def days(self) -> Index:
+        """
+        Number of days for each element.
+        """
+
+        @no_type_check
+        def pandas_days(x) -> int:
+            return x.days
+
+        return ps.Index(self.to_series().transform(pandas_days))
+
+    @property
+    def seconds(self) -> Index:
+        """
+        Number of seconds (>= 0 and less than 1 day) for each element.
+        """
+        sdf = self._internal.spark_frame
+        hour_scol_name = verify_temp_column_name(sdf, "__hour_column__")
+        minute_scol_name = verify_temp_column_name(sdf, "__minute_column__")
+        second_scol_name = verify_temp_column_name(sdf, "__second_column__")
+        sum_scol_name = verify_temp_column_name(sdf, "__sum_column__")
+
+        # Extract the hours part, minutes part, seconds part and its fractional part with microseconds
+        sdf = sdf.select(
+            F.expr("date_part('HOUR', %s)" % SPARK_DEFAULT_INDEX_NAME),
+            F.expr("date_part('MINUTE', %s)" % SPARK_DEFAULT_INDEX_NAME),
+            F.expr("date_part('SECOND', %s)" % SPARK_DEFAULT_INDEX_NAME),
```

Comment:
Hm .. the index name might not be `SPARK_DEFAULT_INDEX_NAME` ... right?
I think you can still do all of them without using `DataFrame.select`.  Can we directly use `F.when(...).otherwise()`  like you did with `self.to_series().transform`? We can just replace `hour_scol_name`, `minute_scol_name` and `second_scol_name` directly to `F.expr(...)` without `DataFrame.select`.

Reply:
For `SPARK_DEFAULT_INDEX_NAME`, I think we might have to directly call `org.apache.spark.sql.catalyst.expressions.DatePart` to invoke Spark column. Feel free to add a function at `PythonSQLUtils` such as:

```scala
// This is only used in Python side to pass `Column` instances with other expressions
// instead of plain column names.
def datePart(field: Column, source: Column, child: Column): Column =
  Column(DatePart(field.expr, source.expr, child.expr))
```

and call it in Python side like:

```python
jvm.PythonSQLUtils.castTimestampNTZToLong(functions.lit('HOUR')._jc, self.spark.column._jc)
```

Reply:
We can add the function in `.../pandas/spark/functions.py` and use `_call_udf` in it

Reply:
Thank you @HyukjinKwon and @ueshin ! Updated.